Studio-Lovelies/GG-JointJustice-Unity
# Feature/20 appearing dialog

## Summary

Created script that makes letters appear one at a time to the screen. You are able to change the speed of punctuation, dialog, or overall speed from the inky file. The code has function ready to speed up the dialog when player gives input, and this feature can be disabled/enabled from the inky file. You can prolong the dialog with next one after players input, or make the dialogs skip to next one without players input.

Does not change the speakers name or background color yet. For this I would request new PR implementing Scriptable Object containing the name, color etc. information about the speaker, and this could be used to changed all information on the screen.
 
## Before/after screenshots and/or animated gif

Example of the dialog appearing
![AppearingDialog](https://user-images.githubusercontent.com/16340242/131007291-c351d506-285a-48e2-98a7-23bb58e73603.gif)


## Testing instructions

Open AppearingText-TestScene and hit play. The textbox contains Lorem Ipsum, but after you press space, the dialog starts. Take a look at the AppearingDialogInk-file while testing the dialog to see how they affect the way dialog appears.

## Additional information

- `[x]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource:  #20
